packages/react/src/components/NumberFieldBase/NumberFieldBase.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { useFocusRing } from '@react-aria/focus';
+import { useHover } from '@react-aria/interactions';
+import { mergeProps, mergeRefs } from '@react-aria/utils';
+import type { FocusableProps, ValidationState } from '@react-types/shared';
+import { cx } from '@project44-manifest/react-styles';
+import { As, createComponent, Options, Props } from '../../system';
+import type { StyleProps } from '../../types';
+import { FormControl } from '../FormControl';
+import { useStyles } from './NumberFieldBase.styles';
+
+export type NumberFieldBaseElement = 'div';
+
+export interface NumberFieldBaseOptions<T extends As = NumberFieldBaseElement>
+  extends Options<T>,
+    FocusableProps,
+    StyleProps {
+  /**
+   * Helper text to append to the form control input element.
+   */
+  helperText?: React.ReactNode;
+  /**
+   * Props passed to the helper text.
+   */
+  helperTextProps?: React.HTMLAttributes<HTMLElement>;
+  /**
+   * The ref passed to the input element.
+   */
+  inputRef?: React.RefObject<HTMLInputElement | HTMLTextAreaElement>;
+  /**
+   * Props passed to the input element.
+   */
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+  /**
+   * Whether the input is disabled.
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether the input can be selected but not changed by the user.
+   */
+  isReadOnly?: boolean;
+  /**
+   * Whether user input is required on the input before form submission.
+   */
+  isRequired?: boolean;
+  /**
+   * Label of the input element
+   */
+  label?: React.ReactNode;
+  /**
+   * Props passed to the label.
+   */
+  labelProps?: React.LabelHTMLAttributes<HTMLLabelElement>;
+  /**
+   * Whether the input should display its "valid" or "invalid" visual styling.
+   */
+  validationState?: ValidationState;
+}
+
+export type NumberFieldBaseProps<T extends As = NumberFieldBaseElement> = Props<
+  NumberFieldBaseOptions<T>
+>;
+
+export const NumberFieldBase = createComponent<NumberFieldBaseOptions>((props, forwardedRef) => {
+  const {
+    autoFocus,
+    className: classNameProp,
+    css,
+    helperText,
+    helperTextProps,
+    inputProps = {},
+    inputRef,
+    isDisabled,
+    isRequired,
+    label,
+    labelProps,
+    validationState,
+  } = props;
+
+  const isInvalid = validationState === 'invalid';
+
+  const fieldRef = React.useRef<HTMLInputElement | HTMLTextAreaElement>(null);
+
+  const { focusProps, isFocused, isFocusVisible } = useFocusRing({
+    autoFocus,
+    isTextInput: true,
+  });
+  const { isHovered, hoverProps } = useHover({ isDisabled });
+
+  const { className } = useStyles({
+    isDisabled,
+    isFocused,
+    isFocusVisible,
+    isHovered,
+    isInvalid,
+    css,
+  });
+
+  const classes = cx(className, classNameProp, {
+    'manifest-textfield-base': true,
```

Comment:
I did it this way (in consultation with @dmorton-p44) in order to have the same styling for `textField` and `numberField` input components, so that `numberField` practically inherits styling of the existing `textField`. You can see in `NumberField.styles.ts` and `NumberFieldBase.styles.ts` I did:
```
export const useStyles = css({ ...TextFieldStyles });
export const useStyles = css({ ...TextFieldBaseStyles });
```
This ways I wanted to:
- Inherit / keep the same styling for components
- Reduce code (styling) duplication

I am aware we are overjumping naming pattern, but just let me know if you think my assumption to reduce duplication and have kind of style inheritance is not stronger then naming pattern. For the sake of simplicity, understanding and debugging naming pattern is definitely better option.
It's not big change, just wasn't sure what approach should be more important.

Reply:
I hear your point and there is a tradeoff, but let's update these strings to strictly match the component, e.g. `manifest-numberfield-base`. To clarify, these class names _do nothing_ for styles. Their sole purpose is to be available for overriding the styles when consumed in micro-webapps, e.g. a dev can target the `manifest-numberfield-base` with some CSS to override the styles built in. 

As for inheriting styles from TextFieldStyles, yes awesome job -- this code is great and definitely the best architecture to reduce defining more styles 
```
export const useStyles = css({ ...TextFieldStyles });
....
```